src/ansys/geometry/core/geometry/curves/trimmed_curve.py

```diff
@@ -0,0 +1,209 @@
+        The start point of the curve.
+    end : Point3D
+        The end point of the curve.
+    interval : Interval
+        The parametric interval that bounds the curve.
+    length : Quantity
+        The length of the curve.
+    grpc_client : GrpcClient, default: None
+        Optional gRPC client that is required for advanced functions such as `intersect_curve()`.
+    """
+
+    def __init__(
+        self,
+        geometry: Curve,
+        start: Point3D,
+        end: Point3D,
+        interval: Interval,
+        length: Quantity,
+        grpc_client: GrpcClient = None,
+    ):
+        """Initialize ``TrimmedCurve`` class."""
+        self._geometry = geometry
+        self._start = start
+        self._end = end
+        self._interval = interval
+        self._length = length
+        self._grpc_client = grpc_client
+        if grpc_client is not None:
+            self._commands_stub = CommandsStub(self._grpc_client.channel)
+
+    @property
+    def geometry(self) -> Curve:
+        """The underlying mathematical curve."""
+        return self._geometry
+
+    @property
+    def start(self) -> Point3D:
+        """The start point of the curve."""
+        return self._start
+
+    @property
+    def end(self) -> Point3D:
+        """The end point of the curve."""
+        return self._end
+
+    @property
+    @protect_grpc
+    def length(self) -> Quantity:
+        """Calculated length of the edge."""
+        return self._length
+
+    @property
+    @protect_grpc
+    def interval(self) -> Interval:
+        """Interval of the curve that provides its boundary."""
+        return self._interval
+
+    def evaluate_proportion(self, param: Real) -> CurveEvaluation:
+        """
+        Evaluate the curve at a proportional value.
+
+        A parameter of 0 would correspond to the start of the curve, while a parameter of 1 would
+        correspond to the end of the curve.
+
+        Parameters
+        ----------
+        param : Real
+            Parameter in the proportional range [0,1].
+
+        Returns
+        -------
+        CurveEvaluation
+            The resulting curve evaluation.
+        """
+        bounds = self.interval
+        return self.geometry.evaluate(bounds.start + bounds.get_span() * param)
+
+    def intersect_curve(self, other: "TrimmedCurve") -> List[Point3D]:
+        """
+        Intersect this trimmed curve with another to receive the points of intersection.
+
+        If the curves do not intersect, an empty list is returned.
+
+        Parameters
+        ----------
+        other : TrimmedCurve
+            The trimmed curve to intersect with.
+
+        Returns
+        -------
+        List[Point3D]
+            All points of intersection between the curves.
+        """
+        if self._grpc_client is None:
+            raise ValueError(
+                """This TrimmedCurve was not initialized with a grpc client,
+                so this method cannot be called."""
```

Comment:
```suggestion
                """Because this trimmed curve was not initialized with a gRPC client,
                the method cannot be called."""
```